lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -519,26 +524,35 @@ func (c *Client) sharedDirectoryInfoRequest(req tdp.SharedDirectoryInfoRequest)
 	return C.ErrCodeSuccess
 }
 
-// close frees the memory of the cgo.Handle,
-// closes the RDP client connection,
-// and frees the Rust client.
+// close closes the RDP client connection and
+// the TDP connection to the browser.
 func (c *Client) close() {
 	c.closeOnce.Do(func() {
 		// Ensure the RDP connection is closed
 		if errCode := C.close_rdp(c.rustClient); errCode != C.ErrCodeSuccess {
 			c.cfg.Log.Warningf("error closing the RDP connection")
 		}
+		c.cfg.Log.Debug(("RDP connection closed successfully"))
+
+		// Ensure the TDP connection is closed
+		if err := c.cfg.Conn.Close(); err != nil {
+			c.cfg.Log.Warningf("error closing the TDP connection: %v", err)
+		}
+		c.cfg.Log.Debug(("TDP connection closed successfully"))
+	})
+}
 
+// cleanup frees the Rust client and
+// frees the memory of the cgo.Handle.
+func (c *Client) cleanup() {
+	c.cleanupOnce.Do(func() {
```

Comment:
Do we need to use a once here? Now that this is no longer a race between two goroutines, I think we can probably get rid of it.

Reply:
We don't need it necessarily as written, but I added it so that it can never be used in the future in a way that causes the program to panic (which may happen if `C.free_rdp(c.rustClient)` is called twice, and will definitely happen if `c.handle.Delete()` is called twice).

Reply:
Sure, but by that logic you should have a once for `Run` as well, etc.

I think protecting against multiple calls is fine if it is an exported method, as the calling code may do something wrong.

In this case, it's all internal so my vote is remove the once.